### DISCUSSION TITLE
Agent logging policy

### DISCUSSION THREAD

**Discussion by gsmirnov-splk** [open-telemetry/opentelemetry-java-instrumentation#4886] [Q&A]
Hi team, is there any generally accepted logging policy for otel instrumentation agents?

I understand that this is a trade-off between not polluting application logs and capturing useful agent info. For example, I was recently troubleshooting an agent connectivity issue. Until I enabled debug logs, the agent had logged zero complaints. With debug logs, the problem became immediately clear: the plain HTTP endpoint URL was specified with `https://` protocol. So I created this issue: https://github.com/signalfx/splunk-otel-java/issues/566 that started some discussion around the subject. I'd love to hear more about this.

**Comment by trask** [comment]
hi @gsmirnov-splk!

upstream otel exporters log connectivity issues at warning level (with some throttling)

in the microsoft disto exporter, we log right away on first warning, and then aggregate and log at most every 5 minutes to minimize log spam ([AggregatingLogger.java](https://github.com/microsoft/ApplicationInsights-Java/blob/main/agent/agent-tooling/src/main/java/com/microsoft/applicationinsights/agent/internal/common/AggregatingLogger.java))

**Reply by gsmirnov-splk** [reply]
@trask got it, thanks! Are there any generally accepted guidelines for this across vendors? Who would be the right person to ask this?

**Reply by iNikem** [reply]
I don't think there are any recommendations about SDKs' or auto-instrumentations' logging behaviour in the spec. Certainly not among vendors. You may try to raise this question in spec repo.
